lib/screens/change_email.dart

```diff
@@ -0,0 +1,167 @@
+        if (_fetching_email) {
+          return LinearProgressIndicator();
+        } else {
+          return Container(
+            child: Form(
+              key: _formKey,
+              child: ListView(
+                padding: EdgeInsets.all(20),
+                children: <Widget>[buildEmailField(), submitButton()],
+              ),
+            ),
+          );
+        }
+      }),
+    );
+  }
+
+  Widget buildEmailField() {
+    return TextFormField(
+      validator: (String value) {
+        if (value.isEmpty ||
+            !RegExp(r"[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?")
+                .hasMatch(value)) {
+          return 'Please enter valid email address';
+        }
+      },
+      controller: _textFieldController,
+      decoration: InputDecoration(
+        labelText: "Email",
+      ),
+      onSaved: (String value) {
+        setState(() {
+          _email = value;
+        });
+      },
+    );
+  }
+
+  Widget submitButton() {
+    return ScopedModelDescendant(
+        builder: (BuildContext context, Widget child, MainModel model) {
+      return FlatButton(
+          color: Colors.green,
+          disabledColor: Colors.grey,
+          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
+          child: Text(
+            'SAVE EMAIL',
+            style: TextStyle(color: Colors.white),
+          ),
+          onPressed: _saving_email
+              ? null
+              : () async {
+                  saveEmail(context, model);
+                });
+    });
+  }
+
+  saveEmail(context, model) async {
+    setState(() {
+      _saving_email = true;
+    });
+    Map<dynamic, dynamic> updateResponse;
+    if (!_formKey.currentState.validate()) {
+      return;
+    }
+    _formKey.currentState.save();
+    final SharedPreferences prefs = await SharedPreferences.getInstance();
+    String user_id = prefs.getInt('id').toString();
+    Map<String, String> headers = Map();
+    headers = {
+      'Content-Type': 'application/json',
+      'token-type': 'Bearer',
+      'ng-api': 'true',
+      'Auth-Token': prefs.getString('spreeApiKey')
+    };
+    Map<String, dynamic> payload = Map();
+    payload = {'email': _email, 'user_id': user_id};
+    String url = Settings.SERVER_URL + "api/v1/users/${user_id}";
+    http.Response response =
+        await http.put(url, headers: headers, body: json.encode(payload));
+    updateResponse = json.decode(response.body);
+    setState(() {
+      _saving_email = false;
+    });
+    logoutUser(context, model);
+  }
+
+  logoutUser(BuildContext context, MainModel model) async {
+    final SharedPreferences prefs = await SharedPreferences.getInstance();
+    String user_id = prefs.getInt('id').toString();
+    String api_key = prefs.getString('spreeApiKey');
+    Map<String, String> headers = {
```

Comment:
Please use the common getHeaders() method in headers.dart in order to avoid ambiguity.